force-app/main/default/classes/ReceiptTriggerHandler.cls

```diff
@@ -0,0 +1,83 @@
+                                                                                                                                                    FROM Cases
+                                                                                                                                                    WHERE Id IN:listCasesIdsFromReceipts)
+                                     FROM Contact ];
+
+        Map<Id, Contact> doctorsMap = new Map<Id, Contact>();
+        List<Id> doctorsDepartmentsIdsList = new List<Id>();
+
+        for (Contact contact_i : doctorsList){
+            doctorsMap.put(contact_i.Id, contact_i);
+            doctorsDepartmentsIdsList.add(contact_i.AccountId);
+        }
+
+        List<AggregateResult> allDoctorsTreatmentsList = [SELECT SUM(Amount_of_payments__c)amount, ContactId ctId
+                                                          FROM Case 
+                                                          WHERE ContactId IN:doctorsMap.keySet() AND CreatedDate = THIS_MONTH AND CreatedDate = THIS_YEAR
+                                                          GROUP BY ContactId];
+
+        List<Contact> doctorIdsToUpdateList = new List<Contact>();
+        //3.1
+        for (AggregateResult result_i : allDoctorsTreatmentsList){
+            if (doctorsMap.containsKey((Id) result_i.get('ctId'))){
+                if (doctorsMap.get((Id) result_i.get('ctId')).Account.Doctor_payments_to_receive_bonus__c < (Decimal) result_i.get('amount')){
+                    doctorIdsToUpdateList.add(new Contact(Id = (Id) result_i.get('ctId'), Payments_amount__c = (Decimal) result_i.get('amount')));
+                }
+
+            }
+        }
+        List<Database.SaveResult> updateResults = Database.update (doctorIdsToUpdateList, false);
+
+        for (Database.SaveResult result : updateResults){
+            if (result.isSuccess()){
+                System.debug('Successfully published event.');
+            } else{
+                for (Database.Error error : result.getErrors()){
+                    System.debug(error.getMessage());
+                    System.debug(error.getStatusCode());
+                }
+            }
+        }
+
+        //3.2 
+        List<Account> departmentList = [SELECT Id, Name, (SELECT Id, Payments_amount__c, Best_doctor_of_the_month__c
+                                                          FROM Contacts)
+                                        FROM Account 
+                                        WHERE Id IN:doctorsDepartmentsIdsList];
+        System.debug('Departments list :' + departmentList);
+
+            Map<Id, Contact> doctorsToUpdateMap = new Map<Id, Contact> ();
+        
+        for (Account department_i : departmentList) {
+            Id doctorIdWithHighestPayments;
+            Decimal highestAmount = 0;
+            for (Contact contact_i : department_i.Contacts) {
+                if (contact_i.Payments_amount__c > highestAmount) {
```

Comment:
you can Order by Payments_amount__c  in SOQL query

Reply:
I had tried but somthing went wrong. I'll try again. Thnx